components/builder-jobsrv/src/server/worker_manager.rs

```diff
@@ -197,7 +203,13 @@ impl WorkerMgr {
                         continue;
                     }
                 }
-                None => break,
+                None => {
+                    debug!("no workers available - bailing for now");
+                    job.set_state(jobsrv::JobState::Pending);
+                    self.datastore.set_job_state(&job)?;
```

Comment:
I agree with @mwrock, we might want to remove this bit since the default state of a job is already Pending